### ISSUE TITLE
BUG: Application Crash Managing Break Playlist

### ISSUE THREAD

**Issue by SchmalzTech** [OpenKJ/OpenKJ#265]
Hello everyone! I heard of the passing of Mr. Lightburn. I hope we can keep this project alive in his memory! I'm an awful programmer, but I can at least submit a bug report when I find one. I haven't done very many of these, but I will be as complete as possible to document the issue. Windows did dump memory, but I don't want to post unless absolutely necessary in case it would leak any confidential information.

Please note the logs show a corrupt karaoke zip file. This should not be related to the issue. This is logged prior to the steps to cause the crash.

### **Symptom**
Application freezes and crashes. Debug log appears to not log any fault.

### **Steps to Reproduce**
Break DB consists entirely of MP4 video files (I believe encoded H.264, would have to verify.) Create a playlist consisting of MP4 video files, then attempt to drag and drop in the playlist pane to change the order. App crashes every time with multiple playlists consisting of different files. This does not occur when selecting a file in the playlist, then moving it up or down using the arrow buttons at the bottom of the window, nor does the randomize button cause issues.

### **Debug Log (Set to trace level, this is everything logged from app open to crash)**
`
[2022-02-22 17:17:53.213] [info] OpenKJ version 2.0.8 starting up
[2022-02-22 17:17:53.232] [info] App dir path C:/Program Files (x86)/OpenKJ
[2022-02-22 17:17:53.232] [debug] Checking for other instances
[2022-02-22 17:17:53.354] [warning] QSqlQuery::exec: database not open
[2022-02-22 17:17:53.355] [warning] QSqlQuery::exec: database not open
[2022-02-22 17:17:53.905] [debug] [MediaBackend] [KAR] Constructing GStreamer backend
[2022-02-22 17:17:53.905] [info] [MediaBackend] [KAR] Hardware accelerated video rendering mode: true
[2022-02-22 17:17:53.905] [debug] [MediaBackend] [KAR] Building GStreamer pipeline
[2022-02-22 17:17:53.905] [debug] [MediaBackend] [KAR] Gstreamer not initialized yet, initializing
[2022-02-22 17:17:53.980] [trace] [KARFader] setVolumeElement called
[2022-02-22 17:17:53.995] [info] [MediaBackend] [KAR] Using SoundTouch pitch shifter
[2022-02-22 17:17:53.996] [debug] [MediaBackend] [KAR] Setting volume to: 0
[2022-02-22 17:17:53.996] [info] [MediaBackend] [KAR] Setting audio output device to default
[2022-02-22 17:17:53.996] [debug] [MediaBackend] [KAR] Unlinking and removing old elements
[2022-02-22 17:17:53.996] [debug] [MediaBackend] [KAR] Creating new audio sink element
[2022-02-22 17:17:53.996] [debug] [MediaBackend] [KAR] Adding and linking new audio output element
[2022-02-22 17:17:53.996] [debug] [MediaBackend] [KAR] Setting volume to: 0
[2022-02-22 17:17:53.996] [debug] [MediaBackend] [KAR] Gstreamer pipeline build completed
[2022-02-22 17:17:54.044] [info] [MediaBackend] [KAR] Setting audio output device to "1 - Speaker (SB X-Fi Go! Pro)"
[2022-02-22 17:17:54.044] [debug] [MediaBackend] [KAR] Unlinking and removing old elements
[2022-02-22 17:17:54.044] [debug] [MediaBackend] [KAR] Creating new audio sink element
[2022-02-22 17:17:54.044] [debug] [MediaBackend] [KAR] Adding and linking new audio output element
[2022-02-22 17:17:54.044] [debug] [MediaBackend] [KAR] GStreamer backend construction complete
[2022-02-22 17:17:54.045] [debug] [MediaBackend] [SFX] Constructing GStreamer backend
[2022-02-22 17:17:54.045] [info] [MediaBackend] [SFX] Hardware accelerated video rendering mode: true
[2022-02-22 17:17:54.045] [debug] [MediaBackend] [SFX] Building GStreamer pipeline
[2022-02-22 17:17:54.045] [trace] [SFXFader] setVolumeElement called
[2022-02-22 17:17:54.049] [debug] [MediaBackend] [SFX] Setting volume to: 0
[2022-02-22 17:17:54.049] [info] [MediaBackend] [SFX] Setting audio output device to default
[2022-02-22 17:17:54.049] [debug] [MediaBackend] [SFX] Unlinking and removing old elements
[2022-02-22 17:17:54.049] [debug] [MediaBackend] [SFX] Creating new audio sink element
[2022-02-22 17:17:54.049] [debug] [MediaBackend] [SFX] Adding and linking new audio output element
[2022-02-22 17:17:54.050] [debug] [MediaBackend] [SFX] Setting volume to: 0
[2022-02-22 17:17:54.050] [debug] [MediaBackend] [SFX] Gstreamer pipeline build completed
[2022-02-22 17:17:54.057] [info] [MediaBackend] [SFX] Setting audio output device to "0 - Default"
[2022-02-22 17:17:54.057] [debug] [MediaBackend] [SFX] Unlinking and removing old elements
[2022-02-22 17:17:54.057] [debug] [MediaBackend] [SFX] Creating new audio sink element
[2022-02-22 17:17:54.057] [debug] [MediaBackend] [SFX] Adding and linking new audio output element
[2022-02-22 17:17:54.058] [debug] [MediaBackend] [SFX] GStreamer backend construction complete
[2022-02-22 17:17:54.058] [debug] [MediaBackend] [BM] Constructing GStreamer backend
[2022-02-22 17:17:54.058] [info] [MediaBackend] [BM] Hardware accelerated video rendering mode: true
[2022-02-22 17:17:54.058] [debug] [MediaBackend] [BM] Building GStreamer pipeline
[2022-02-22 17:17:54.058] [trace] [BMFader] setVolumeElement called
[2022-02-22 17:17:54.063] [debug] [MediaBackend] [BM] Setting volume to: 0
[2022-02-22 17:17:54.063] [info] [MediaBackend] [BM] Setting audio output device to default
[2022-02-22 17:17:54.063] [debug] [MediaBackend] [BM] Unlinking and removing old elements
[2022-02-22 17:17:54.063] [debug] [MediaBackend] [BM] Creating new audio sink element
[2022-02-22 17:17:54.063] [debug] [MediaBackend] [BM] Adding and linking new audio output element
[2022-02-22 17:17:54.064] [debug] [MediaBackend] [BM] Setting volume to: 0
[2022-02-22 17:17:54.064] [debug] [MediaBackend] [BM] Gstreamer pipeline build completed
[2022-02-22 17:17:54.070] [info] [MediaBackend] [BM] Setting audio output device to "0 - Default"
[2022-02-22 17:17:54.070] [debug] [MediaBackend] [BM] Creating new audio sink element
[2022-02-22 17:17:54.071] [debug] [MediaBackend] [BM] Adding and linking new audio output element
[2022-02-22 17:17:54.070] [debug] [MediaBackend] [BM] Unlinking and removing old elements
[2022-02-22 17:17:54.071] [debug] [MediaBackend] [BM] GStreamer backend construction complete
[2022-02-22 17:17:54.072] [info] [AudioRecorder] Initializing AudioRecorder instance
[2022-02-22 17:17:54.072] [debug] [AudioRecorder] initGStreamer() called
[2022-02-22 17:17:54.083] [debug] [AudioRecorder] Elements created, adding to pipeline and linking
[2022-02-22 17:17:54.084] [debug] [AudioRecorder] initGStreamer() completed
[2022-02-22 17:17:54.105] [warning] QObject::connect(IdleDetect, OKJSongbookAPI): invalid nullptr parameter
[2022-02-22 17:17:54.315] [info] [MainWindow] Database schema version: 106
[2022-02-22 17:17:54.460] [info] [KaraokeSongsModel] Loaded 0 karaoke songs from the db on disk
[2022-02-22 17:17:54.472] [debug] [RotationModel] loading rotation data from DB on disk
[2022-02-22 17:17:54.473] [debug] [RotationModel] loaded 0 rotation singers
[2022-02-22 17:17:54.627] [info] [KaraokeSongsModel] Loaded 0 karaoke songs from the db on disk
[2022-02-22 17:17:54.642] [info] tsWidth =  97
[2022-02-22 17:17:54.642] [info] singerColSize =  93
[2022-02-22 17:17:54.644] [info] [SongShop] Requesting songs from db.openkj.org
[2022-02-22 17:17:54.673] [trace] [TickerImageCreator] Thread starting up
[2022-02-22 17:17:54.673] [info] [TickerImageCreator] Rendering ticker text: No ticker data
[2022-02-22 17:17:54.675] [info] [TickerDisplayWidget] Enabled set to: true
[2022-02-22 17:17:54.676] [info] [TickerThread] Ticker starting
[2022-02-22 17:17:54.676] [trace] [TickerImageCreator] Thread starting up
[2022-02-22 17:17:54.676] [info] [TickerImageCreator] Rendering ticker text: No ticker data
[2022-02-22 17:17:54.696] [trace] [TickerImageCreator] Ticker image rendered in 23ms
[2022-02-22 17:17:54.697] [trace] [TickerImageCreator] Ticker image rendered in 20ms
[2022-02-22 17:17:54.757] [info] [MainWindow] Adding singer count to status bar
[2022-02-22 17:17:55.455] [info] Requesting current version info for branch:  "stable"
[2022-02-22 17:17:55.466] [info] [MainWindow] Initial UI setup complete
[2022-02-22 17:17:55.547] [info] [MainWindow] Resizing videoPreview to width: 300
[2022-02-22 17:17:55.551] [info] [LazyDurationController] Finding songs with missing durations
[2022-02-22 17:17:55.554] [info] [LazyDurationController] Done, found 1 songs with missing durations
[2022-02-22 17:17:55.554] [info] [LazyDurationThread] Starting scan
[2022-02-22 17:17:55.654] [info] [MediaBackend] [BM] Stop requested
[2022-02-22 17:17:55.654] [debug] [MediaBackend] [BM] Backend already in stopped state, aborting
[2022-02-22 17:17:55.654] [debug] [MediaBackend] [BM] Setting volume to: 50
[2022-02-22 17:17:55.654] [debug] [MediaBackend] [KAR] Setting volume to: 99
[2022-02-22 17:17:55.654] [debug] [MainWindow] Initial volumes - K: 99 - BM 50
[2022-02-22 17:17:55.656] [debug] [MediaBackend] [KAR] Turning on silence detection
[2022-02-22 17:17:55.656] [debug] [MediaBackend] [BM] Turning on silence detection
[2022-02-22 17:17:55.685] [debug] [RotationModel] Setting singer id: -1 name: 'invalid' as the current rotation singer
[2022-02-22 17:17:55.686] [info] [PlaylistSongsModel] Loaded 13 songs from db on disk
[2022-02-22 17:17:55.796] [trace] [ExternalZipUtil] Infozip output: 
Archive:  M:/Karaoke/14416 - H13555-You Can Do It - Ice Cube, Mack 10 and Ms Toi.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
 18888500  01/22/2013 00:40   H13555 - You Can Do It - Ice Cube - Mack 10 - Ms Toi.mp4
---------                     -------
 18888500                     1 file

[2022-02-22 17:17:55.796] [error] [ExternalZipUtil] Error processing infozip output, bailing out
[2022-02-22 17:17:55.796] [info] [LazyDurationThread] Scan complete
[2022-02-22 17:17:55.796] [warning] [LazyDurationThread] Unable to get duration for file M:/Karaoke/14416 - H13555-You Can Do It - Ice Cube, Mack 10 and Ms Toi.zip. - File is likely corrupted or invalid
[2022-02-22 17:17:55.951] [info] Received network reply
[2022-02-22 17:17:55.951] [info] Received version:  "2.0.8"  Current version:  "2.0.8"
[2022-02-22 17:17:56.084] [trace] [MainWindow] [rotationDataChanged] Called
[2022-02-22 17:17:56.084] [trace] [MainWindow] [rotationDataChanged] finished in 0ms
[2022-02-22 17:17:56.084] [info] [MainWindow] Resizing videoPreview to width: 300
[2022-02-22 17:17:56.085] [trace] [TickerImageCreator] Thread starting up
[2022-02-22 17:17:56.085] [info] [TickerImageCreator] Rendering ticker text: Redacted Ticker Text - LOL • Singers: 0 • Current: None
[2022-02-22 17:17:56.383] [trace] [TickerImageCreator] Ticker image rendered in 298ms
[2022-02-22 17:17:59.470] [info] "[\"{32547008-fc69-4830-b456-1f77c6857fd0}\",\"release\",\"2.0.8\",\"winnt\",\"Windows 10 Version 2009\",\"x86_64\",\"WAN IP address redacted\"]"
[2022-02-22 17:18:00.950] [trace] [SongShop] Received network reply from db.openkj.org
`

### **Windows Event Log Entry**
`
Log Name:      Application
Source:        Application Error
Date:          2/22/2022 5:18:02 PM
Event ID:      1000
Task Category: (100)
Level:         Error
Keywords:      Classic
User:          N/A
Computer:      Karaoke-PC
Description:
Faulting application name: openkj.exe, version: 0.0.0.0, time stamp: 0x6140eff8
Faulting module name: Qt5Core.dll, version: 5.15.2.0, time stamp: 0x5fa4dd3b
Exception code: 0xc0000005
Fault offset: 0x0000000000211a58
Faulting process id: 0x2590
Faulting application start time: 0x01d8283a0900a50a
Faulting application path: C:\Program Files (x86)\OpenKJ\openkj.exe
Faulting module path: C:\Program Files (x86)\OpenKJ\Qt5Core.dll
Report Id: 79070da1-732f-4034-9324-0185e348cfbc
Faulting package full name: 
Faulting package-relative application ID: 
Event Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Application Error" />
    <EventID Qualifiers="0">1000</EventID>
    <Version>0</Version>
    <Level>2</Level>
    <Task>100</Task>
    <Opcode>0</Opcode>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2022-02-22T22:18:02.7513273Z" />
    <EventRecordID>16052</EventRecordID>
    <Correlation />
    <Execution ProcessID="0" ThreadID="0" />
    <Channel>Application</Channel>
    <Computer>Karaoke-PC</Computer>
    <Security />
  </System>
  <EventData>
    <Data>openkj.exe</Data>
    <Data>0.0.0.0</Data>
    <Data>6140eff8</Data>
    <Data>Qt5Core.dll</Data>
    <Data>5.15.2.0</Data>
    <Data>5fa4dd3b</Data>
    <Data>c0000005</Data>
    <Data>0000000000211a58</Data>
    <Data>2590</Data>
    <Data>01d8283a0900a50a</Data>
    <Data>C:\Program Files (x86)\OpenKJ\openkj.exe</Data>
    <Data>C:\Program Files (x86)\OpenKJ\Qt5Core.dll</Data>
    <Data>79070da1-732f-4034-9324-0185e348cfbc</Data>
    <Data>
    </Data>
    <Data>
    </Data>
  </EventData>
</Event>
`

### **OS Info**
Edition	Windows 10 Pro
Version	21H1
OS build	19043.1526
Experience	Windows Feature Experience Pack 120.2212.4170.0

**Comment by SchmalzTech** [comment]
I built 2.0.8 release from source on an Ubuntu 20.04.4 machine, and I did not experience the issue. May be isolated to Windows builds.

**Comment by SchmalzTech** [comment]
Further troubleshooting on another Windows device, I could not replicate the issue. I have yet to compare Windows patch/build level between the two, and I plan to back up then delete any OpenKJ configuration files and the database on the affected device to see if starting from scratch resolves the issue. The affected device has had several previous versions installed prior to v2.0.8-release. I will update this issue after these steps.

**Comment by SchmalzTech** [comment]
Offending computer is as follows:
Intel Core2 Quad Q9500
12GB RAM
Windows 10 Pro 21H1 19043.1586

I might close this issue since it seems to be isolated to this one computer, at least in my experience. Perhaps someone can replicate the issue?

**Comment by SchmalzTech** [comment]
I am going to close this. I have tried further and cannot replicate this issue on any other computer with any OS. I may need to purge my configuration files and reinstall on the problem machine (which is unfortunately my production machine.)

**Comment by SchmalzTech** [comment]
RESOLUTION: I did not narrow it down to a specific setting, but I renamed my openkj.ini file so it generated a new one with defaults and the crash went away. I had almost everything customized in my installation, including the font and font sizes. I may step through the differences between that old config and the current working one to try to narrow this down. This config file was created under a much older version of the software, so I may not be able to replicate the issue in a new file.
